header/testing.go

```diff
@@ -171,15 +177,15 @@ func RandExtendedHeader(t *testing.T) *ExtendedHeader {
 }
 
 // RandRawHeader provides a RawHeader fixture.
-func RandRawHeader(t *testing.T) *RawHeader {
+func RandRawHeader(t *testing.T, dataHash []byte) *RawHeader {
```

Comment:
nit: `RandRawHeader` is meant to really be rand, so it can be used in tests that don't care about data hashes or other fields. This change makes `RandRawHeader` more oriented toward data testing, thus making it not credibly neutral :joy:, so I would just set DataHash field on the returned RawHeader instead.

Reply:
In this PR in `TestSuites`'s `GenRawHeader` we pass the dataHash and then assign its field another time btw

Reply:
ok